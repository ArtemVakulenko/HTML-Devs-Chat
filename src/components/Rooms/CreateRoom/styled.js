import styled from 'styled-components';

export const StCreateRoomContainer = styled.div`
  display: flex;
  justify-content: center;
  align-content: center;
`;
