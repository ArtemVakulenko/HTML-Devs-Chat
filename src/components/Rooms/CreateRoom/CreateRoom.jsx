import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../../_common/Button';
import CreateRoomModal from '../CreateRoomModal';
import { StCreateRoomContainer } from './styled.js';

const CreateRoom = ({ getAllUsers }) => {
  const [modalHide, setModalHide] = useState(false);
  const openModal = () => {
    setModalHide(true);
    getAllUsers();
  };
  const closeModal = () => {
    setModalHide(false);
  };
  return (
    <>
      {
        modalHide
          ? <CreateRoomModal onClose={closeModal} />
          : null
      }
      <StCreateRoomContainer >
        <Button
          value="createRoom"
          onClick={openModal}
        />
      </StCreateRoomContainer>
    </>
  );
};

export default CreateRoom;

CreateRoom.propTypes = {
  getAllUsers: PropTypes.func.isRequired,
};
