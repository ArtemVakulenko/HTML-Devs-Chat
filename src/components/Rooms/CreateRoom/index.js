import { connect } from 'react-redux';
import { getAllUsers } from '../../../store/modals/actions';
import CreateRoom from './CreateRoom';

const mapDispatchToProps = {
  getAllUsers,
};
export default connect(null, mapDispatchToProps)(CreateRoom);
