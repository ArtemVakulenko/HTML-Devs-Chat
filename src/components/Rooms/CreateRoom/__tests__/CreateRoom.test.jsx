import React from 'react';
import CreateRoom from '../CreateRoom';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));

describe('CreateRoom', () => {
  const props = {
    getAllUsers: jest.fn(),
  };
  it('Should match snapshot', () => {
    const component = global.shallow(<CreateRoom {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have Button', () => {
    const component = global.mount(<CreateRoom {...props} />);
    expect(component.find('Button')).toHaveLength(1);
  });
  it('Should have StCreateRoomContainer', () => {
    const component = global.mount(<CreateRoom {...props} />);
    expect(component.find('styled__StCreateRoomContainer')).toHaveLength(1);
  });
});
