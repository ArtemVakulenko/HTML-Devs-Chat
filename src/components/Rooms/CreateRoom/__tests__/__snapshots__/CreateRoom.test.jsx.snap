// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoom Should match snapshot 1`] = `"<div class=\\"styled__StCreateRoomContainer-hJkdEc daenTo\\"><button class=\\"styled__StButton-iwFefC fohYxd\\"></button></div>"`;
